modules/apps/frontend-js/frontend-js-web/src/main/java/com/liferay/frontend/js/web/internal/resource/JavaScriptFrontendResource.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.frontend.js.web.internal.resource;
+
+import com.liferay.petra.io.StreamUtil;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.util.ContentTypes;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import java.net.URL;
+
+import java.nio.charset.StandardCharsets;
+
+import java.util.ResourceBundle;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+public class JavaScriptFrontendResource implements FrontendResource {
+
+	public JavaScriptFrontendResource(
+		String eTag, boolean immutable, Language language, long maxAge,
+		ResourceBundle resourceBundle, boolean sendNoCache, URL url) {
+
+		if (resourceBundle != null) {
+			if (eTag != null) {
+				throw new IllegalArgumentException(
+					"Translated resources cannot have an eTag");
+			}
+
+			if (immutable) {
+				throw new IllegalArgumentException(
+					"Translated resources cannot be immutable");
+			}
+		}
+
+		_eTag = eTag;
+		_immutable = immutable;
+		_language = language;
+		_maxAge = maxAge;
+		_resourceBundle = resourceBundle;
+		_sendNoCache = sendNoCache;
+		_url = url;
+	}
+
+	@Override
+	public String getContentType() {
+		return ContentTypes.APPLICATION_JAVASCRIPT;
+	}
+
+	@Override
+	public String getETag() {
+		return _eTag;
+	}
+
+	@Override
+	public InputStream getInputStream() throws IOException {
+		if (_resourceBundle == null) {
+			return _url.openStream();
```

Comment:
So basically, lack of resourceBundle for the resource means no runtime translations. But presence of resourceBundle is determined by existence of `translate` parameter. In this case the resource bundle is returned (with aggregation from portlet resources or not). Is this understanding correct?